plugins/services/src/js/reducers/serviceForm/Containers.js

```diff
@@ -456,19 +411,15 @@ module.exports = {
       return container;
     });
 
-    if (field === 'healthChecks') {
-      if (newState[index].healthChecks == null) {
-        newState[index].healthChecks = [];
-      }
-
-      if (this.healthChecks[index] == null) {
-        this.healthChecks[index] = {};
+    if (field === 'healthCheck') {
+      if (this.healthCheckState[index] == null) {
+        this.healthCheckState[index] = {};
```

Comment:
Yes, in general we need to split it up into multiple reducers, that we combine instead of this mega-file of logic